PennMobileShared/Dining/DiningAnalyticsViewModel.swift

```diff
@@ -194,7 +209,10 @@ public class DiningAnalyticsViewModel: ObservableObject {
         return balanceDiff / timeDiff
     }
     func predictZeroDate(firstBalance: DiningAnalyticsBalance, lastBalance: DiningAnalyticsBalance, slope: Double) -> Date {
-        let offset = -firstBalance.balance / slope
+        var offset = -firstBalance.balance / slope
+        if slope == 0 {
```

Comment:
@JHawk0224 correct me if I'm wrong but I think you can also do:

return slope == 0 ? Date.distantFuture : Calendar.current.date(byAdding: .day, value: Int(offset), to: firstBalance.date)!

Reply:
Not a swift guru lol but I think it still has the ternary operator syntax